### PR TITLE
Adding objects necessary for mapping Oracle schemas and tables to PostgreSQL databases and tables

### PR DESCRIPTION
**Pull request by vchske**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json [hunk 1]
```diff
@@ -97,6 +97,120 @@
...
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "connectionInfo",
+        "selectedSchemas"
+      ]
+    },
+    "GetUserTablesOracleTaskOutput": {
```
Comment (hitenjava) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json]: Missing change - Where are these being referred?

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json [hunk 2]
```diff
@@ -592,6 +592,20 @@
           "description": "Parent database of the user"
         }
       }
+    },
+    "QueryTable": {
```
Comment (huang91shu) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json]: Maybe it goes by a different name but I think we already have this object somewhere.  Most of the GetUserTables should return an ITable object

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json [hunk 3]
```diff
@@ -97,6 +97,120 @@
...
+      "type": "object",
+      "description": "Input for the task that gets the list of tables contained within a provided list of Oracle schemas.",
+      "properties": {
+        "connectionInfo": {
+          "$ref": "./Common.json#/definitions/OracleConnectionInfo",
+          "description": "Information for connecting to Oracle source"
+        },
+        "selectedSchemas": {
+          "type": "array",
+          "description": "List of Oracle Schemas for which to collect tables",
```
Comment (huang91shu) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json]: Schemas -> schemas

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json [hunk 4]
```diff
@@ -97,6 +97,120 @@
...
+                "items": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "GetUserTablesOracleTaskInput": {
```
Comment (huang91shu) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json]: I think Tasks needs an entry for the new scenarios.

**Review by huang91shu — CHANGES_REQUESTED** [review]

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json [hunk 5]
```diff
@@ -592,6 +592,20 @@
...
+      "type": "object",
+      "description": "Information about a single table.",
+      "properties": {
+        "hasRows": {
+          "type": "boolean",
+          "description": "Whether the table has any rows"
+        },
+        "name": {
+          "type": "string",
+          "description": "The schema-qualifed table name"
```
Comment (erich-wang) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json]: > qualifed 
is it typo?

**Review by erich-wang — CHANGES_REQUESTED** [review]

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json (hunk 5 above)
Comment (nschonni) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json]:
```suggestion
          "description": "The schema-qualified table name"
```

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json (hunk 2 above)
Comment (vchske) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json]: Found it. It was in TasksCommon.json.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json (hunk 5 above)
Comment (vchske) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json]: Removed

Comment (vchske) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Common.json]: Removed

**Comment by vchske** [comment]
@erich-wang I was told that instead of submitting a new commit for any changes that I should force the new changes onto a single commit. As I am not a git master, do you have guidance on how this is done?

**Comment by nschonni** [comment]
@vchske I think they "squash merge" when landing PRs, so you should be able to just push the additional commit. If you want to cleanup the commits down to 1, you can use a rebase and squash like http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html and then `git push -f` to force the new single commit as the branch

**Review by erich-wang — APPROVED** [review]

**Comment by erich-wang** [comment]
@vchske , please resolve the comments from @huang91shu  and @hitenjava.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json (hunk 4 above)
Comment (vchske) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json]: Done

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json (hunk 1 above)
Comment (vchske) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/OracleAzureDbPostgreSqlSyncTask.json]: Are you referring to the missing parts in Tasks.json? If so, these have been added now.

**Review by huang91shu — APPROVED** [review]

**Comment by vchske** [comment]
@erich-wang Comments have been resolved and the change is approved so I think we are ready for a merge.

**Comment by erich-wang** [comment]
@vchske , I just noticed that you missed the **ARM API Review Checklist**, so just added **WaitForARMFeedback**. Of course we could skip this step if @huang91shu is from ARM review board.

**Comment by vchske** [comment]
@erich-wang We aren't making a breaking change with this. We aren't adding or removing an API endpoint. Based on this, I don't believe an ARM review is necessary. This is essentially the same as https://github.com/Azure/azure-rest-api-specs/pull/5801 and that one also did not require an ARM review.

**Comment by erich-wang** [comment]
@vchske , you're adding properties which falls into below checklist:
> Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories.
> adding/removing APIs.
> adding/removing properties.
> adding/removing API-version.
> adding a new service in Azure.

**Comment by erich-wang** [comment]
@ravbhatnagar, could you help review this PR from ARM perspective?

**Review by ryansbenson — APPROVED** [review]

**Pull request merged**